pypeit/utils.py

```diff
@@ -29,6 +30,241 @@
 from pypeit import msgs
 from pypeit.move_median import move_median
 
+def zero_not_finite(array):
+    """
+    Set the elements of an array to zero which are inf or nan
+
+    Args:
+        array (`np.ndarray`_):
+          An numpy array of arbitrary shape that potentially has nans or infinities
+
+    Returns:
+        new_array (`np.ndarray`_)
+          A copy of the array with the nans and inifinities set to zero
+
+    """
+    not_finite = np.logical_not(np.isfinite(array))
+    new_array = array.copy()
+    new_array[not_finite] = 0.0
+    return new_array
+
+def arr_setup_to_setup_list(arr_setup):
+    """
+
+    Args:
+        arr_setup: (list)
+          A list of length nsetups echelle output arrays of shape=(nspec, norders, nexp)
+
+    Returns:
+        setup_list: (list)
+          List of length nsetups. Each element of the setup list is a list of length
+          norder*nexp elements, each of which contains the shape = (nspec1,) wavelength arrays
+          for the order/exposure in setup1. The list is arranged such that the nexp1 spectra
+          for iorder=0 appear first, then come nexp1 spectra for iorder=1, i.e. the outer or
+          fastest varying dimension in python array ordering is the exposure number.
```

Comment:
I like to delineate the variables returned as per the scikit-learn doc style.

Reply:
but it breaks `readthedocs`

Reply:
It's, of course, fine to delineate the variables returned.  My point was that you're not using the correct syntax to do so.  You have to use Numpy style doc syntax.  Here you're mixing Numpy and Google style, which sphinx will not deal with correctly, and it will not warn you about the syntax errors when you compile the docs.  I'll fix these docs, but it would be very helpful for you to learn the syntax.  See the bare-bones `sphinx.readme` doc I added to the repo.

Reply:
I've pushed a bunch of doc fixes.  I'm still getting errors like:
```
include/links.rst:18: WARNING: Mismatch: both interpreted text role prefix and reference suffix.
```
which I can't track down because the error is unrelated to the `links.rst` file.